packages/google_sign_in/example/lib/main.dart

```diff
@@ -94,18 +97,45 @@ class SignInDemoState extends State<SignInDemo> {
     return null;
   }
 
-  Future<Null> _handleSignIn() async {
-    try {
-      await _googleSignIn.signIn();
-    } catch (error) {
-      print(error);
-    }
+  void _handleSignIn() {
+    _handleGoogleSignIn()
+        .then((GoogleSignInAccount user) => _afterSignIn)
+        .catchError((dynamic error) => _onSignInError);
```

Comment:
I actually prefer that we don't catch anything other than PlatformException. The previous code was also incorrect. A better version would be:

try {
   await signin();
} on PlatformException catch(e) {
  // handle platform exception
}